QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+        isDone = true;
+    } while (isDone == false);
+
+    return 0;
+}
+
+
+int QSPIFBlockDevice::_reset_flash_mem()
+{
+    // Perform Soft Reset of the Device prior to initialization
+    int status = 0;
+    char status_value[2] = {0};
+    QSPIF_LOG(QSPIF_DEBUG_INFO, "INFO: _reset_flash_mem:\n");
+    //Read the Status Register from device
+    if (QSPI_STATUS_OK == _qspi_send_general_command(QSPIF_RDSR, -1, NULL, 0, status_value,
+            1) ) {  // store received values in status_value
+        QSPIF_LOG(QSPIF_DEBUG_DEBUG, "DEBUG: Reading Status Register Success: value = 0x%x\n", (int)status_value[0]);
+    } else {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Reading Status Register failed\n");
+        status = -1;
+    }
+
+    if (0 == status) {
+        //Send Reset Enable
+        if (QSPI_STATUS_OK == _qspi_send_general_command(QSPIF_RSTEN, -1, NULL, 0, NULL,
+                0) ) {   // store received values in status_value
+            QSPIF_LOG(QSPIF_DEBUG_DEBUG, "DEBUG: Sending RSTEN Success\n");
+        } else {
+            QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Sending RSTEN failed\n");
+            status = -1;
+        }
+
+
+        if (0 == status) {
+            //Send Reset
+            if (QSPI_STATUS_OK == _qspi_send_general_command(QSPIF_RST, -1, NULL, 0, NULL,
+                    0)) {   // store received values in status_value
+                QSPIF_LOG(QSPIF_DEBUG_DEBUG, "DEBUG: Sending RST Success\n");
+            } else {
+                QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Sending RST failed\n");
+                status = -1;
+            }
+
+            _is_mem_ready();
+        }
+    }
+
+    return status;
+}
+
+
+bool QSPIFBlockDevice::_is_mem_ready()
+{
+    // Check Status Register Busy Bit to Verify the Device isn't Busy
+    char status_value[2];
+    int retries = 0;
+    bool memReady = true;
+
+    do {
+        retries++;
+        //Read the Status Register from device
+        if (QSPI_STATUS_OK != _qspi_send_general_command(QSPIF_RDSR, -1, NULL, 0, status_value,
+                2)) {   // store received values in status_value
+            QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Reading Status Register failed\n");
+        }
+    } while ( (status_value[0] & 0x1) != 0 && retries < IS_MEM_READY_MAX_RETRIES );
+
+    if ((status_value[0] & 0x1) != 0) {
+        QSPIF_LOG(QSPIF_DEBUG_DEBUG, "ERROR: _is_mem_ready FALSE\n");
+        memReady = false;
+    }
+    return memReady;
+}
+
+
+int QSPIFBlockDevice::_set_write_enable()
+{
+    int status = 0;
+    if (QSPI_STATUS_OK !=  _qspi_send_general_command(QSPIF_WREN, -1, NULL, 0, NULL, 0)) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR:Sending WREN command FAILED\n");
+        status = -1;
+    }
+    return status;
+}
+
+
+
+/*********************************************/
+/************* Utility Functions *************/
+/*********************************************/
+int QSPIFBlockDevice::_utils_find_addr_region(int offset)
```

Comment:
address is defined as `bd_address_t`, so why here is int and then cast everywhere in this file?

Reply:
fixed, thanks